solver/src/solver/http_solver/model.rs

```diff
@@ -26,14 +26,16 @@ pub struct OrderModel {
 }
 
 #[derive(Debug, Serialize)]
-pub struct UniswapModel {
-    pub token1: String,
-    pub token2: String,
-    #[serde(with = "serde_with::rust::display_fromstr")]
-    pub balance1: u128,
-    #[serde(with = "serde_with::rust::display_fromstr")]
-    pub balance2: u128,
+pub struct PoolTokenData {
     #[serde(with = "serde_with::rust::display_fromstr")]
+    pub balance: u128,
+    pub weight: String,
```

Comment:
This shouldn't be `String` - its a 256-bit integer.

Reply:
I spoke with @marcovc and he agreed with string. However I can make it u128 like the balance and to the `#[serde(with = "serde_with::rust::display_fromstr")]` if this is more appropriate here.

Reply:
Actually in the balancer module this is a `Bfp`. Is this something we can include here or should we convert `Bfp`to `u128` when sending it on?

Reply:
So, we can use `Bpf`, but I don't know how easy it would be to support arbitrary precision decimal numbers (**not** double precision floating point numbers) on the solver side (cc @marcovc and @twalth3r). Alternatively, we can use their `as_uint256` to match their EVM representation - whichever is easier for the solver team.

Reply:
I was under the impression that we had been using u128 everywhere else because we couldn't adapt U256 to the solver team's needs. Perhaps we should ask with @e00E, @twalth3r  and/or @marcovc if there was a particular reason for u128 as balance and if we could just start using `U256` here. Otherwise I would prefer to leave it as u128 and convert the `Bfp`to this (if possible).

Reply:
> for u128 as balance and if we could just start using U256 here

Uniswap stores balances in `u96`, so `u128` is the smallest integer capable of representing Uniswap balances.

Reply:
I'm not sure what's going on on the rust side, but we're getting big ints (as in anything > 2^64) as strings: because of json parser limitations.

Reply:
In that case, we should be able to use `U256` with the `u256_decimal` formatter.

Reply:
So in that case, I would then opt for going with U256 in both the balance and weight field here. I hope this is approach is a mutual feeling.